mongoose.c

```diff
@@ -1419,22 +1419,40 @@ void cs_hmac_sha1(const unsigned char *key, size_t keylen,
 
 void mbuf_init(struct mbuf *mbuf, size_t initial_size) WEAK;
 void mbuf_init(struct mbuf *mbuf, size_t initial_size) {
+#if MG_ENABLE_THREADSAFE_MBUF
+    pthread_mutex_init(&(mbuf->mutex), NULL);
+#endif
+
   mbuf->len = mbuf->size = 0;
   mbuf->buf = NULL;
   mbuf_resize(mbuf, initial_size);
+
 }
 
 void mbuf_free(struct mbuf *mbuf) WEAK;
 void mbuf_free(struct mbuf *mbuf) {
   if (mbuf->buf != NULL) {
```

Comment:
What if it gets freed after this check?

Reply:
Actually more importantly mbuf.buf can be null while the mutex is still initialized. So fixed that.. good catch. Fixed the problem in both places. Merging this now :tada: